hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/resources/ResourceHandlerModule.java

```diff
@@ -69,9 +70,9 @@ public class ResourceHandlerModule {
   private static volatile CGroupsHandler cGroupsHandler;
   private static volatile CGroupsBlkioResourceHandlerImpl
       cGroupsBlkioResourceHandler;
-  private static volatile CGroupsMemoryResourceHandlerImpl
+  private static volatile AbstractCGroupsMemoryResourceHandler
```

Comment:
Nit: there might not always be an abstract parent class for every ResourceHandler, instead we could simply return the interface like some of the getters already do.

AbstractCGroupsMemoryResourceHandler -> MemoryResourceHandler
AbstractCGroupsCpuResourceHandler -> CpuResourceHandler

Reply:
Thanks @brumi1024 for the review! Fixed your comment and rebased the patch to the current trunk.